config/config.go

```diff
@@ -160,9 +167,20 @@ func (c *S3Cli) allowMultipart() bool {
 			return false
 		}
 	}
+	if c.isAlicloudHost() {
+		return false
+	}
 	return true
 }
 
 func (c *S3Cli) getRegionFromHost() string {
 	return AWSHostToRegion[c.Host]
 }
+
+func (c *S3Cli) isAlicloudHost() bool {
+	return strings.HasPrefix(c.Host, "oss-") && strings.HasSuffix(c.Host, "aliyuncs.com")
```

Comment:
Would be cool if we shared this logic with the regex recommended below.

Reply:
Thanks, and I will improve it.